src/main/java/it/pagopa/interop/probing/statistics/api/service/impl/StatisticServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package it.pagopa.interop.probing.statistics.api.service.impl;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import it.pagopa.interop.probing.statistics.api.service.StatisticService;
+import it.pagopa.interop.probing.statistics.api.service.TimestreamService;
+import it.pagopa.interop.probing.statistics.api.util.logging.Logger;
+import it.pagopa.interop.probing.statistics.dtos.EserviceStatus;
+import it.pagopa.interop.probing.statistics.dtos.PercentageContent;
+import it.pagopa.interop.probing.statistics.dtos.StatisticContent;
+import it.pagopa.interop.probing.statistics.dtos.StatisticsEserviceResponse;
+
+@Service
+public class StatisticServiceImpl implements StatisticService {
+
+  @Autowired
+  private TimestreamService timestreamService;
+
+  @Autowired
+  private Logger logger;
+
+  @Override
+  public StatisticsEserviceResponse findStatistics(Long eserviceRecordId, Integer pollingFrequency)
+      throws IOException {
+    logger.logRequest(eserviceRecordId, pollingFrequency);
+    List<StatisticContent> content =
+        timestreamService.findStatistics(eserviceRecordId, pollingFrequency);
+    content = cleanStatistics(content);
+    List<PercentageContent> percenteges = calculatePercentages(content);
+    return StatisticsEserviceResponse.builder().values(content).percentages(percenteges).build();
+  }
+
+  private List<StatisticContent> cleanStatistics(List<StatisticContent> content) {
+    List<StatisticContent> falseND = new ArrayList<>();
+    for (StatisticContent statistic : content) {
+      if (statistic.getStatus().equals(EserviceStatus.N_D)) {
+        if (!getNext(statistic, content).getStatus().equals(EserviceStatus.N_D)
+            && !getPrevious(statistic, content).getStatus().equals(EserviceStatus.N_D)) {
+          falseND.add(statistic);
+        }
```

Comment:
is this the part we discussed that should be done with a direct query on timestream? or it is a different query?

Reply:
Basically we fill the gaps created during the query by the polling frequency with the N/D value. Because of the fact that the full circle of operations takes slightly more than the polling frequency time, we will have this gaps filled with N/D which is false. So in case of an isolated N/D we will just ignore it.